### PR TITLE
CI/CD fails on production

### PR DESCRIPTION
**Pull request by antonyharfield**
## ✅ DoD

- [x] All work is complete
- [x] Migrations: NA
- [x] Documentation: NA
- [x] Release notes: NA
- [x] Issues linked: resolves _none_

## 🛑 Problems

_none_

## 📝 Summary

- The build is failing on prod! Renamed to production. :)

## 💡 More ideas

_none_

## 📸 Screenshots

_none_

### CODE REVIEW

File: .github/workflows/build-test-deploy.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
           map: |
             {
               "main": {
-                "BUILD_ALIAS": "prod",
+                "BUILD_ALIAS": "production",
```
Comment (charles-allen) [inline .github/workflows/build-test-deploy.yml:38]: Ahh... I see! The Firestore-backup storage locations are hardcoded against `production`. Probably that should be configuration not in code!! _Edit: See #53 for more ideas_

File: .github/workflows/build-test-deploy.yml [hunk 2]
```diff
@@ -45,7 +45,7 @@ jobs:
                 "SECRET_NAME_FIREBASE_PROJECT_ID": "PROD_FIREBASE_PROJECT_ID",
                 "SECRET_NAME_FIREBASE_STORAGE_BUCKET": "PROD_FIREBASE_STORAGE_BUCKET",
                 "SECRET_NAME_FUNCTIONS_REGION": "PROD_FUNCTIONS_REGION",
-                "VUE_APP_MODE": "PROD"
+                "VUE_APP_MODE": "PRODUCTION"
```
Comment (charles-allen) [inline .github/workflows/build-test-deploy.yml]: We should probably delete this. 
Also in local configs & scripts.

File: .github/workflows/build-test.yml [hunk 3]
```diff
@@ -25,45 +25,6 @@ jobs:
...
-                "SECRET_NAME_FUNCTIONS_REGION": "DEV_FUNCTIONS_REGION",
-                "VUE_APP_MODE": "DEV"
-              },
-              ".*": {
-                "VUE_APP_EMU_PORT_AUTH": "",
-                "VUE_APP_EMU_PORT_FIRESTORE": "",
-                "VUE_APP_EMU_PORT_FUNCTIONS": ""
-              }
-            }
-
```
Comment (charles-allen) [inline .github/workflows/build-test.yml:66]: Good catch

File: .github/workflows/build-test.yml [hunk 4]
```diff
@@ -85,15 +46,17 @@ jobs:
...
-          VUE_APP_FIREBASE_API_KEY: ${{ secrets[env.SECRET_NAME_FIREBASE_API_KEY] }}
-          VUE_APP_FIREBASE_APP_ID: ${{ secrets[env.SECRET_NAME_FIREBASE_APP_ID] }}
-          VUE_APP_FIREBASE_AUTH_DOMAIN: ${{ secrets[env.SECRET_NAME_FIREBASE_AUTH_DOMAIN] }}
-          VUE_APP_FIREBASE_DATABASE_U_R_L: ${{ secrets[env.SECRET_NAME_FIREBASE_DATABASE_U_R_L] }}
-          VUE_APP_FIREBASE_MESSAGING_SENDER_ID: ${{ secrets[env.SECRET_NAME_FIREBASE_MESSAGING_SENDER_ID] }}
-          VUE_APP_FIREBASE_PROJECT_ID: ${{ secrets[env.SECRET_NAME_FIREBASE_PROJECT_ID] }}
-          VUE_APP_FIREBASE_STORAGE_BUCKET: ${{ secrets[env.SECRET_NAME_FIREBASE_STORAGE_BUCKET] }}
-          VUE_APP_FUNCTIONS_REGION: ${{ secrets[env.SECRET_NAME_FUNCTIONS_REGION] }}
-          VUE_APP_BUILD_VERSION: ${{ env.BUILD_VERSION }}
+          BUILD_ALIAS: production
```
Comment (charles-allen) [inline .github/workflows/build-test.yml]: For `vue-cli-service build`, the [default mode is already `production`](https://v3.cli.vuejs.org/guide/mode-and-env.html#modes).

...and I think it's irrelevant because I don't think we have *any* runtime code that's dependent on mode. It only drives which `.env` files to pick up; but on GitHub you explicitly delete `.env` & set it via CLI.

TL;DR: You can delete the `BUILD_ALIAS` (in this script) & remove `--mode...` if you want.

File: .github/workflows/build-test.yml [hunk 5]
```diff
@@ -85,15 +46,17 @@ jobs:
...
+          BUILD_VERSION: ${{ steps.package-version.outputs.current-version }}
+          VUE_APP_FIREBASE_API_KEY: abcd
+          VUE_APP_FIREBASE_APP_ID: 1234
+          VUE_APP_FIREBASE_AUTH_DOMAIN: xyz
+          VUE_APP_FIREBASE_DATABASE_U_R_L: xyz
+          VUE_APP_FIREBASE_MESSAGING_SENDER_ID: 1234
+          VUE_APP_FIREBASE_PROJECT_ID: xyz
+          VUE_APP_FIREBASE_STORAGE_BUCKET: xyz
+          VUE_APP_FUNCTIONS_REGION: xyz
+          VUE_APP_BUILD_VERSION: PR
```
Comment (charles-allen) [inline .github/workflows/build-test.yml]: Good idea!

**Review by charles-allen — APPROVED** [review]
If you want, there are a couple of _vestigial_ lines of code that could be deleted:
- All refs to `VUE_APP_MODE`
- Setting `--mode` for GitHub build (it only affects `.env` pickup locally)

Also happy to leave it as a low-priority PBI for later.

File: .github/workflows/build-test-deploy.yml (hunk 2 above)
Comment (antonyharfield) [reply .github/workflows/build-test-deploy.yml]: I have removed it everywhere

**Comment by charles-allen** [comment]
@antonyharfield - You earned this:
clap
